game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,155 @@
+zuus_cloud_oaa = class( AbilityBaseClass )
+LinkLuaModifier( "modifier_zuus_cloud_oaa", "abilities/oaa_zuus_cloud.lua", LUA_MODIFIER_MOTION_NONE )
+
+function zuus_cloud_oaa:OnSpellStart()
+  local caster = self:GetCaster()
+  local hCloud = CreateUnitByName( "npc_dota_zeus_cloud", self:GetCursorPosition(), true, caster, caster, caster:GetTeamNumber() )
+  hCloud:SetOwner( self:GetCaster() )
+  hCloud:SetControllableByPlayer( self:GetCaster():GetPlayerOwnerID(), false )
+  hCloud:AddNewModifier( caster, self, "modifier_zuus_cloud_oaa", nil )
+  hCloud:AddNewModifier( caster, self, "modifier_kill", { duration = self:GetSpecialValueFor( "cloud_duration" ) } )
+  FindClearSpaceForUnit( hCloud, self:GetCursorPosition(), true )
+end
+
+function zuus_cloud_oaa:OnHeroCalculateStatBonus()
+	local caster = self:GetCaster()
+
+	if caster:HasScepter() then
+		self:SetHidden( false )
+		if self:GetLevel() <= 0 then
+			self:SetLevel( 1 )
+		end
+	else
+		self:SetHidden( true )
+	end
+end
+
+modifier_zuus_cloud_oaa = class( ModifierBaseClass )
+
+function modifier_zuus_cloud_oaa:IsHidden()
+  return true
+end
+
+function modifier_zuus_cloud_oaa:IsDebuff()
+  return false
+end
+
+function modifier_zuus_cloud_oaa:IsPurgable()
+  return false
+end
+
+function modifier_zuus_cloud_oaa:RemoveOnDeath()
+  return true
+end
+
+function modifier_zuus_cloud_oaa:OnCreated( kv )
+  if IsServer() then
+    self.Interval = self:GetAbility():GetSpecialValueFor("cloud_bolt_interval")
+    self.Radius = self:GetAbility():GetSpecialValueFor( "cloud_radius" )
+    self.cloud_particle = ParticleManager:CreateParticle("particles/units/heroes/hero_zeus/zeus_cloud.vpcf", PATTACH_ABSORIGIN, self:GetParent())
+    ParticleManager:SetParticleControl(self.cloud_particle, 1, Vector(self.Radius, 1, 1))
+
+    ParticleManager:SetParticleControlEnt(self.cloud_particle, 2, self:GetParent(), PATTACH_POINT_FOLLOW, "attach_hitloc", self:GetParent():GetAbsOrigin(), true)
+    self:StartIntervalThink( 0.1 )
+    EmitSoundOn("Hero_Zuus.Cloud.Cast", self:GetParent())
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnDestroy()
+  if IsServer() then
+    ParticleManager:DestroyParticle(self.cloud_particle, false)
+    ParticleManager:ReleaseParticleIndex(self.cloud_particle)
+    self.cloud_particle = nil
+  end
+end
+
+function modifier_zuus_cloud_oaa:DeclareFunctions()
+	local funcs =
+	{
+		MODIFIER_PROPERTY_INCOMING_DAMAGE_PERCENTAGE
```

Comment:
Also, super minor detail, but the damage numbers when Nimbus is getting hit seem to always show 1. Not necessary to replicate that of course, but maybe it could give some idea as to how it's normally done in vanilla.

Reply:
I did not understand your first comment, Can you re phrase?
regarding the damage number, I have no Idea why and how to change, any idea?

Reply:
Damage number can't really be manipulated directly, it's just a side effect of how exactly the damage was manipulated.

What I meant with the first comment is that if ever there's any modifier that also uses the incoming damage percentage property that applies to the Nimbus, it would break the hits-to-kill functionality as it would stack additively. I'm unsure what is the best way, but I think using the absolute no damage properties and then reducing health OnAttacked like someone in the ModDota Discord suggested might be a good way.